### ISSUE TITLE
Add target word in a sentence 

### ISSUE THREAD

**Issue by bee-san** [donkuri/lapis#14]
Currently there are 3 types of cards:
* Click cards
* Sentence Cards
* Hint Cards

I am currently using sentence cards like this:
![image](https://github.com/user-attachments/assets/de98f944-db7e-4123-a5d8-01627e76f88a)

Where the vocab is bolded. The css for bolded vocab does not appear to work in the sentence, like this:
![image](https://github.com/user-attachments/assets/c30319d3-e1b7-4271-98e9-e00ded1a0a24)

You will see "nai" (forgive me for romaji) is bolded here in Anki, but on the card it is not bolded.

I presume this is meant to happen.

JPMN has a concept called target sentence card where the word is clearly bolded in the sentence:
![image](https://github.com/user-attachments/assets/d6eccbe1-b555-4847-8c7d-22e008758cd5)

I prefer this over sentence cards so I can read the outer context of the sentence to understand my vocab.

# Suggestion

I am not sure how this works under the hood, but I am thinking:
* New card type `_isTargetSentenceCard`
* This card type is a copy and paste of the `sentence` type
* This card applies the CSS to bold the word in the sentence

**Comment by bee-san** [comment]
PS: I love the cream colour, it looks beautiful on this card!

**Comment by bee-san** [comment]
Tell you what, I have some time off next week.

@donkuri If you want this in Lapis, mind assigning it to me so I can take a look at adding it? :D

**Comment by YonKuma** [comment]
Huh, that's odd. My sentence cards do have the target word bolded on them. Is it possible you're using (or defaulting back to) a font that doesn't have boldface support?

**Comment by bee-san** [comment]
TL;DR tested on 3 platforms, IOS and a non-addon anki and I still cannot get bolding to work. The rest of this post is my small exploration into why

![2024_11_21_16-02__cMU](https://github.com/user-attachments/assets/e75135b8-6905-4d2c-bcc4-6e4bf8ebf6fe)

On lightmode it uses this CSS, which sets it to this hex colour for bold. 

![2024_11_21_16-04__5Le](https://github.com/user-attachments/assets/4b8b9d6b-a87c-472a-9bd2-ab8b123235e9)

I manually changed the HTML to add the `<b>` tag, and it does apper bolded now.

I changed it to black and reset the bolding
![2024_11_21_16-05__MmA](https://github.com/user-attachments/assets/8ecc8f8e-191f-464f-84e5-8a2a9fc80410)

Again, `bold` is set to a specific hex colour but it obviously isn't bolded.

I tested this on IOS, and then installed Anki on a device with no addons and still get the same issue.

![2024_11_21_16-07__r5v](https://github.com/user-attachments/assets/5fb88244-7fed-4a55-af77-697abef611de)

Interestingly the HTML uses the same class for all of the terms, and does not indicate which term is the term we want to bold. I am not sure how you would auto-bold this in the code? 🤔 

It is bolded in Anki, and Anki probably passes that to your card template. But somewhere along the line, this boldness is removed :(

**Comment by bewizible** [comment]
@bee-san Hello! Ruri here, I made this card. Really nice to see you like the cream color, I do too! (light mode user at times, guilty as charged). I feel like the problem you have stems from one of these two things. Can you see the raw html for the `Sentence` and `SentenceFurigana` field? The word you are trying to bold **must** be enclosed in `<b>` tags, much like this picture

![image](https://github.com/user-attachments/assets/19290efd-f363-4106-9fca-209aba9ba0d1)

I'd like you to check this out. However I think the problem might be this. Now on the second part. These are mostly the same as @YonKuma mentioned. I think your `SentenceFurigana` field **does not** have the `<b>` tags required, I highly think you should generate furigana through AJT Japanese with the `Sentence` already having the correct bolding through Yomitan. Please check through these and let me know!

**Comment by bewizible** [comment]
@bee-san Also, I think your Anki fields might be using Smplified Chinese hànzì fonts, not Japanese fonts 😅  Your 食 and郎 definitely look Chinese

**Comment by bee-san** [comment]
> Can you open up the HTML Editor for your Sentence field? The contents should look like: `<b>このやろー</b>おまえの`. I'm guessing yours may be doing something different.
> 
> Also, do you have anything in your Sentence Furigana field? If so, how is that being generated?

Thanks! Furigana was the issue :)

**Comment by bee-san** [comment]
> d Chinese hànzì fonts, not Japanese fonts 😅 Your 食 and郎 definitely look Chinese

You are right!

![image](https://github.com/user-attachments/assets/76f760c8-a967-4618-9ccb-cd37a084e233)

I have the language pack installed and it looks fine in Chrome, but in Anki it's wrong....

**Comment by bee-san** [comment]
![image](https://github.com/user-attachments/assets/4cd9a434-b61b-4c15-a49a-fd6e38b776bc)

Okay so I think in the editor / Anki window it shows Chinese font, but on Chrome and when repping the card it's a Japanese font....

**Comment by donkuri** [comment]
Hey @bee-san, if the issue is only in the browser and not on the card, I think it's probably fine. I wonder if there's a way to fix this though. If there is, please let me know so that I can add it to [my website](https://donkuri.github.io/learn-japanese/).
Cheers :)
